### ISSUE TITLE
Gitlab Scan job fail for master branch scan

### ISSUE THREAD

**Issue by vlsecurity** [checkmarx-ltd/cx-flow#1188]
### Description

checkmarx-scan gitlab job fails with error 

> Active Scan with Id 1020781 already exists for Project: 48

This scanid was created by this job itself and no other scan was performed during the job. 

According to checkmarx logs the scan has been started at 16:29, but the job found existingScanId at 16:26:57, that means it was created by the job.  For me it looks like cx-flow creates a scan and than checks that the scan already exists and crashes.
Also the scan created is `INCREMENTAL`, but it is FALSE in gitlab-ci config.
scan-queuing and  scan-resubmit are set to FALSE.
In case scan-queuing is true it creates 2 scans. One Incremental and second is usual.

It happens on each scan and looks like a bug. I would appreciate any help and comments if you face the same issue.

### Expected Behavior
Scan job doesn't crash

### Reproduction
Pre-step: At least one scan of main branch should present and project is created in Checkmarx SAST.

Create gitlab-ci using the [template](https://github.com/checkmarx-ltd/cx-flow/blob/develop/templates/gitlab/v3/Checkmarx.gitlab-ci.yml) for Checkmarx and merge into master branch to run the `checkmarx-scan` job
Version: cx-flow-1.6.39
Product: Checkmarx SAST Engine
-	Main Version: 9.5.3.1001

### Log

```
 ${CX_FLOW_EXE} --scan --app="${CI_PROJECT_NAME}" --namespace="${CI_PROJECT_NAMESPACE}" --repo-name="${CI_PROJECT_NAME}" --repo-url="${CI_REPOSITORY_URL}" --cx-team="${CX_TEAM}" --cx-project="${CX_PROJECT}" --branch="${CI_COMMIT_BRANCH}" --spring.profiles.active="${CX_FLOW_ENABLED_VULNERABILITY_SCANNERS}" --f=. ${PARAMS}
 ....
 2023-02-16 16:26:49.325  INFO 12 --- [           main] c.c.f.CxFlowRunner                        [] : =======BUILD INFO=======
2023-02-16 16:26:49.326  INFO 12 --- [           main] c.c.f.CxFlowRunner                        [] : Version: cx-flow-1.6.39
2023-02-16 16:26:49.327  INFO 12 --- [           main] c.c.f.CxFlowRunner                        [] : Time: 2023-02-15T11:39:38.585Z
2023-02-16 16:26:49.328  INFO 12 --- [           main] c.c.f.CxFlowRunner                        [] : =======================
2023-02-16 16:26:50.179  INFO 12 --- [           main] c.c.f.CxFlowApplication                   [] : Started CxFlowApplication in 7.497 seconds (JVM running for 8.573)
2023-02-16 16:26:50.193  INFO 12 --- [           main] c.c.f.CxFlowRunner                        [YGeQvMRt] : Using custom bean implementation  for bug tracking
2023-02-16 16:26:50.250  INFO 12 --- [           main] c.c.f.s.ScaFilterFactory                  [YGeQvMRt] : Initializing SCA filters.
2023-02-16 16:26:50.251  INFO 12 --- [           main] c.c.f.CxFlowRunner                        [YGeQvMRt] : Executing scan process
2023-02-16 16:26:50.252  INFO 12 --- [           main] c.c.f.s.ScaFilterFactory                  [YGeQvMRt] : Initializing SCA filters.
2023-02-16 16:26:50.259  INFO 12 --- [           main] c.c.f.s.ProjectNameGenerator              [YGeQvMRt] : Project name being used: project-master
2023-02-16 16:26:50.261  INFO 12 --- [           main] c.c.f.u.ZipUtils                          [YGeQvMRt] : Creating zip file **/.*.zip from contents of path .
2023-02-16 16:26:50.261  INFO 12 --- [           main] c.c.f.u.ZipUtils                          [YGeQvMRt] : Applying exclusions: .jar
2023-02-16 16:26:56.203  INFO 12 --- [           main] c.c.f.u.ZipUtils                          [YGeQvMRt] : Successfully created **/.*.zip 
2023-02-16 16:26:56.204  INFO 12 --- [           main] c.c.f.s.ScanRequestConverter              [YGeQvMRt] : Overriding team with /CxServer/project
2023-02-16 16:26:56.211  INFO 12 --- [           main] c.c.s.s.CxAuthService                     [YGeQvMRt] : Logging into Checkmarx https://url/cxrestapi/auth/identity/connect/token
2023-02-16 16:26:57.145  INFO 12 --- [           main] c.c.s.s.CxService                         [YGeQvMRt] : Retrieving Cx teams
2023-02-16 16:26:57.192  INFO 12 --- [           main] c.c.s.s.CxService                         [YGeQvMRt] : Found team /CxServer/project with ID 8
2023-02-16 16:26:57.832  INFO 12 --- [           main] c.c.f.s.AbstractVulnerabilityScanner      [YGeQvMRt] : Checking if there is any existing scan for Project: 48
2023-02-16 16:26:57.910  INFO 12 --- [           main] c.c.s.s.CxService                         [YGeQvMRt] : Scan status is 10 for Project: 48
2023-02-16 16:26:57.910  INFO 12 --- [           main] c.c.f.s.AbstractVulnerabilityScanner      [YGeQvMRt] : ----------existingScanId---------: 1020781
2023-02-16 16:26:57.910  WARN 12 --- [           main] c.c.f.s.AbstractVulnerabilityScanner      [YGeQvMRt] : Property scan-resubmit set to false : New scan not submitted, due to existing ongoing scan for the same Project id 48
2023-02-16 16:26:57.921 ERROR 12 --- [           main] c.c.f.s.AbstractVulnerabilityScanner      [YGeQvMRt] : CheckmarxException: Active Scan with Id 1020781 already exists for Project: 48
com.checkmarx.sdk.exception.CheckmarxException: Active Scan with Id 1020781 already exists for Project: 48
	at com.checkmarx.flow.service.AbstractVulnerabilityScanner.executeCxScan(AbstractVulnerabilityScanner.java:2[73](https://***/-/jobs/3877980#L73))
	at com.checkmarx.flow.service.AbstractVulnerabilityScanner.scanLocalPath(AbstractVulnerabilityScanner.java:322)
	at com.checkmarx.flow.service.AbstractVulnerabilityScanner.scanCli(AbstractVulnerabilityScanner.java:201)
	at com.checkmarx.flow.CxFlowRunner.lambda$scanCommon$0(CxFlowRunner.java:623)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:1[75](https://***/-/jobs/3877980#L75))
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:546)
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:505)
	at com.checkmarx.flow.CxFlowRunner.runOnActiveScanners(CxFlowRunner.java:695)
	at com.checkmarx.flow.CxFlowRunner.scanCommon(CxFlowRunner.java:623)
	at com.checkmarx.flow.CxFlowRunner.scanLocalPath(CxFlowRunner.java:603)
	at com.checkmarx.flow.CxFlowRunner.commandLineRunner(CxFlowRunner.java:479)
	at com.checkmarx.flow.CxFlowRunner.run(CxFlowRunner.java:94)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:762)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at com.checkmarx.flow.CxFlowApplication.main(CxFlowApplication.java:21)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)
2023-02-16 16:26:57.979  INFO 12 --- [           main] jsonLogger                                [YGeQvMRt] : 
2023-02-16 16:26:57.998 ERROR 12 --- [           main] c.c.f.s.AbstractVulnerabilityScanner      [YGeQvMRt] : Error occurred in scanLocalPath method.
com.checkmarx.flow.exception.MachinaException: Checkmarx Error Occurred: CheckmarxException: Active Scan with Id 1020781 already exists for Project: 48
	at com.checkmarx.flow.service.AbstractVulnerabilityScanner.executeCxScan(AbstractVulnerabilityScanner.java:295)
	at com.checkmarx.flow.service.AbstractVulnerabilityScanner.scanLocalPath(AbstractVulnerabilityScanner.java:322)
	at com.checkmarx.flow.service.AbstractVulnerabilityScanner.scanCli(AbstractVulnerabilityScanner.java:201)
	at com.checkmarx.flow.CxFlowRunner.lambda$scanCommon$0(CxFlowRunner.java:623)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:546)
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:505)
	at com.checkmarx.flow.CxFlowRunner.runOnActiveScanners(CxFlowRunner.java:695)
	at com.checkmarx.flow.CxFlowRunner.scanCommon(CxFlowRunner.java:623)
	at com.checkmarx.flow.CxFlowRunner.scanLocalPath(CxFlowRunner.java:603)
	at com.checkmarx.flow.CxFlowRunner.commandLineRunner(CxFlowRunner.java:479)
	at com.checkmarx.flow.CxFlowRunner.run(CxFlowRunner.java:94)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:[76](https://***/-/jobs/3877980#L76)2)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at com.checkmarx.flow.CxFlowApplication.main(CxFlowApplication.java:21)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)
2023-02-16 16:26:57.[99](https://***/-/jobs/3877980#L99)9  INFO 12 --- [           main] c.c.f.CxFlowRunner                        [YGeQvMRt] : Finished with exit code: 3
```

**Comment by satyamchaurasiapersistent** [comment]
If two scan runs in parallel this issue occurs in case of same branch and project. Please check in SAST if for same branch and same project scan is finished then re-run pipeline it will work.
